addons/mass_mailing/wizard/test_mailing.py

```diff
@@ -35,8 +35,6 @@ def send_mail_test(self, cr, uid, ids, context=None):
                     'mailing_id': mailing.id,
                 }
                 mail_mail_obj = Mail.browse(cr, uid, Mail.create(cr, uid, mail_values, context=context), context=context)
-                unsubscribe_url = Mail._get_unsubscribe_url(cr, uid, mail_mail_obj, test_mail, context=context)
```

Comment:
Because we will have Un-subscription link already in mail body now. Also in the previous code, there was a line 
body = tools.append_content_to_html(mailing.body_html, unsubscribe_url, plaintext=False, container_tag='p') . 
But "body" variable wasn't used any-where and that link was never been sent actually in test mail.